packages/cinterop/src/nativeDarwinTest/kotlin/io/realm/kotlin/test/CinteropTest.kt

```diff
@@ -79,6 +82,36 @@ import kotlin.test.assertTrue
 // These test are not thought as being exhaustive, but is more to provide a playground for
 // experiments and maybe more relevant for reproduction of C-API issues.
 class CinteropTest {
+    /**
+     * Tests whether our autorelease pointer wrapper releases native memory.
+     *
+     * Allocates a Realm pointer wrapped with our GC autorelease wrapper, then returns the reference
+     * to the releasable pointer that would tell if the underlying pointer has been released.
+     */
+    @Test
+    fun cpointerWrapper_releasesWhenGCed() {
+        val releasablePointer = {
+            memScoped {
+                val realmSchemaNew = realm_schema_new(
+                    classes = allocArray(0),
+                    num_classes = 0u,
+                    class_properties = allocArray(0)
+                )
+
+                CPointerWrapper<RealmSchemaT>(realmSchemaNew)._ptr
+            }
+        }()
+
+        // The pointer has not been reclaimed
+        assertFalse(releasablePointer.released.value)
+
+        // Trigger GC and wait for some time to allow it to collect the object
+        GC.collect()
+        platform.posix.sleep(5u)
```

Comment:
How stable is this? In other places, we have put in loops that test this a few times just to avoid flaky tests. I would probably recommend doing the same here.